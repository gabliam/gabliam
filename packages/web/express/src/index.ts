import { ExpressPlugin } from './express-plugin';

export * from 'http-status-codes';
export * from './express';
export * from './express-converter';
export * from './express-interceptor';
export * from './interfaces';
export * from './request';
export * from './response';
export { ExpressPlugin as default };
