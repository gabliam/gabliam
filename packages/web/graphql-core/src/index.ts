export { TYPE } from './constants';
export * from './graphql-core-plugin';
export * from './interfaces';
export * from './resolver-decorator';
export {
  Arg,
  ArgOptions,
  Args,
  ArgsType,
  Authorized,
  createParamDecorator,
  createMethodDecorator,
  Ctx,
  Directive,
  Extensions,
  registerEnumType,
  Field,
  FieldOptions,
  FieldResolver,
  Info,
  InputType,
  InputTypeOptions,
  InterfaceType,
  InterfaceTypeOptions,
  Mutation,
  ObjectType,
  ObjectTypeOptions,
  PubSub,
  Query,
  Resolver,
  Root,
  Subscription,
  SubscriptionOptions,
  createUnionType,
  UseMiddleware,
  ID, Int, ClassType, Publisher
} from 'type-graphql';
