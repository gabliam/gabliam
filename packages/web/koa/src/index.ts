import { KoaPlugin } from './koa-plugin';

export * from 'http-status-codes';
export * from './interfaces';
export * from './koa';
export * from './koa-converter';
export * from './koa-interceptor';
export * from './request';
export * from './response';
export { KoaPlugin as default };
