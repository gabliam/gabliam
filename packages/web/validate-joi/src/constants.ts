export const METADATA_KEY = {
  validate: '_validate',
};

export const ERRORS_MSGS = {
  DUPLICATED_VALIDATE_DECORATOR: `Cannot apply @Validate decorator multiple times.`,
};
