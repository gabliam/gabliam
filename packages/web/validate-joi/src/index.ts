export * from './metadatas';
export * from './validate';
export * from './constants';
export * from './validate-config';
export { ValidatePlugin as default } from './validate-plugin';
