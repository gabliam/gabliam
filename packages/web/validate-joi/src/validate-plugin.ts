import { Plugin, Scan } from '@gabliam/core';

@Plugin()
@Scan()
export class ValidatePlugin {}
