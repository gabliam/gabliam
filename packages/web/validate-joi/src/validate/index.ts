export * from './validate-interceptor';
export * from './validate-request';
