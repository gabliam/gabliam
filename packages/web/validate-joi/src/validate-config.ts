import { Config, Init } from '@gabliam/core';
import { WebConfiguration } from '@gabliam/web-core';
import { ValidateSendErrorInterceptor, ValidateInterceptor } from './validate';

@Config()
export class ValidateConfig {
  constructor(private webConfiguration: WebConfiguration) {}

  @Init()
  init() {
    this.webConfiguration.useGlobalInterceptor(
      ValidateSendErrorInterceptor,
      ValidateInterceptor
    );
  }
}
