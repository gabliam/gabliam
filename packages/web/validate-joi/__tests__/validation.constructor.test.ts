import { Joi, ValueExtractor } from '@gabliam/core';
import ExpressPlugin from '@gabliam/express';
import KoaPlugin from '@gabliam/koa';
import { Controller, Get, RestController } from '@gabliam/web-core';
import { WebPluginTest } from '@gabliam/web-core/src/testing';
import { Validate, Validator, ValidatorOptionsConstructor } from '../src';

let appTest: WebPluginTest;

[
  {
    name: 'express',
    plugin: ExpressPlugin,
  },
  {
    name: 'koa',
    plugin: KoaPlugin,
  },
].forEach((plugin) => {
  describe(`${plugin.name} integration`, () => {
    beforeEach(async () => {
      appTest = new WebPluginTest([plugin.plugin]);
      appTest.addConf('app.accept', /xml/);
    });

    afterEach(async () => {
      await appTest.destroy();
    });

    [
      {
        name: 'RestController',
        deco: RestController,
      },
      {
        name: 'Controller',
        deco: Controller,
      },
    ].forEach((decorator) => {
      describe(`${decorator.name} validations constructor`, () => {
        const construct = (
          value: ValueExtractor,
        ): Validator | ValidatorOptionsConstructor => ({
          validator: {
            headers: {
              accept: Joi.string().regex(value('app.accept', 'json')),
            },
          },
          options: { allowUnknown: true },
        });

        it('req.headers', async () => {
          @decorator.deco('/')
          class TestController {
            @Get('/')
            @Validate(construct)
            async getTest() {
              return 'lol';
            }
          }

          appTest.addClass(TestController);
          await appTest.buildAndStart();
          const response = await appTest
            .supertest()
            .get('/')
            .set('Accept', 'application/json')
            .expect(400);
          if (decorator.name === 'Controller') {
            expect(response.status).toBe(400);
            expect(response.text).toBeDefined();
          } else {
            expect(response).toMatchSnapshot();
          }

          const response2 = await appTest
            .supertest()
            .get('/')
            .set('Accept', 'application/json')
            .expect(400);
          if (decorator.name === 'Controller') {
            expect(response2.status).toBe(400);
            expect(response2.text).toBeDefined();
          } else {
            expect(response2).toMatchSnapshot();
          }
        });
      });
    });
  });
});
