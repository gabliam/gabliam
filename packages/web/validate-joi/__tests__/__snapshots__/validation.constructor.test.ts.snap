// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express integration RestController validations constructor req.headers 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "190",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"/\\"accept/\\" with value /\\"application/json/\\" fails to match the required pattern: /xml/\\",\\"validation\\":{\\"source\\":\\"headers\\",\\"keys\\":[\\"accept\\"]}}",
}
`;

exports[`express integration RestController validations constructor req.headers 2`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "190",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"/\\"accept/\\" with value /\\"application/json/\\" fails to match the required pattern: /xml/\\",\\"validation\\":{\\"source\\":\\"headers\\",\\"keys\\":[\\"accept\\"]}}",
}
`;

exports[`koa integration RestController validations constructor req.headers 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "190",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"/\\"accept/\\" with value /\\"application/json/\\" fails to match the required pattern: /xml/\\",\\"validation\\":{\\"source\\":\\"headers\\",\\"keys\\":[\\"accept\\"]}}",
}
`;

exports[`koa integration RestController validations constructor req.headers 2`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "190",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"/\\"accept/\\" with value /\\"application/json/\\" fails to match the required pattern: /xml/\\",\\"validation\\":{\\"source\\":\\"headers\\",\\"keys\\":[\\"accept\\"]}}",
}
`;
